src/AssociativeOpsTable.cpp

```diff
@@ -249,6 +249,29 @@ void populate_ops_table_single_uint32_select(const vector<Type> &types, vector<A
     table.emplace_back(select(x0 < -y0, y0, tmax_0), zero_0, true);          // Saturating add
 }
 
+Expr is_nan_not_strict(Expr x) {
```

Comment:
is_nan without strict_float can legally simplify to false, so this code threw me at first. But I realize what you're writing here is not an expression with a value - it's a pattern you're trying to pattern-match an expression against, and you don't want the strict_float to be part of the pattern because you don't want to strip it. If that's correct, please add a clarifying comment.

Reply:
Done